daemon/info_unix.go

```diff
@@ -25,16 +25,18 @@ func (daemon *Daemon) fillPlatformInfo(v *types.Info, sysInfo *sysinfo.SysInfo)
 		v.CgroupVersion = "2"
 	}
 
-	v.MemoryLimit = sysInfo.MemoryLimit
-	v.SwapLimit = sysInfo.SwapLimit
-	v.KernelMemory = sysInfo.KernelMemory
-	v.KernelMemoryTCP = sysInfo.KernelMemoryTCP
-	v.OomKillDisable = sysInfo.OomKillDisable
-	v.CPUCfsPeriod = sysInfo.CPUCfs
-	v.CPUCfsQuota = sysInfo.CPUCfs
-	v.CPUShares = sysInfo.CPUShares
-	v.CPUSet = sysInfo.Cpuset
-	v.PidsLimit = sysInfo.PidsLimit
+	if v.CgroupDriver != cgroupNoneDriver {
+		v.MemoryLimit = sysInfo.MemoryLimit
```

Comment:
Why does `sysInfo` return `true`? Shouldn't the fix be done there if it's returning incorrect info?

Reply:
@AkihiroSuda ^^

Reply:
Because `sysInfo` is not aware of the cgroup driver.

We could modify (in a separate PR) `sysInfo` to return true only when we can grant write permission, the value needs to be negated to false when `v.CgroupDriver==cgroupNoneDriver`.